[🔀 Utility] 팔로우 기능 구현

Cualone opened the pull request:
### 📅 2024.11.12

### 🌵 Branch
feature/follow/#11 → develop

### 📢 Description
 Target 유저를 팔로우하기 (Source 유저가 수행가능)
 Source 유저가 Target 유저를 언팔로우 상태로 변경
(Source 유저 및 Target 유저 양쪽에 의해 수행 가능)
 특정 사용자에 대한 팔로우 상태 조회
 특정 유저의 팔로워 목록 조회
 특정 유저의 팔로잉 목록 조회

### 💬 Issue Number
- #11 

### ✔️ PR Checklist
PR이 다음 요구 사항을 충족하는지 확인하세요.
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 코드가 정상적으로 동작하는지 테스트 했습니다.

### 🔖 Note

Cualone committed:
:recycle: refactor: response, exception 수정

Cualone committed:
:sparkles: feat: Target 유저를 팔로우하기 (Source 유저가 수행)

Cualone committed:
:sparkles: feat: Source 유저가 Target 유저를 언팔로우 상태로 변경

Source 유저 및 Target 유저 양쪽에 의해 수행 가능

Cualone committed:
:sparkles: feat: 특정 사용자에 대한 팔로우 상태 조회

Cualone committed:
:recycle: refactor: error code 수정

Cualone committed:
:twisted_rightwards_arrows: merge: Merge branch 'feature/view/#8' of https://github.com/mul-meong/utility into feature/follow/#11

# Conflicts:
#	src/main/java/com/mulmeong/utility/UtilityApplication.java
#	src/main/java/com/mulmeong/utility/adapter/in/web/controller/DislikeController.java
#	src/main/java/com/mulmeong/utility/adapter/in/web/controller/LikesController.java
#	src/main/java/com/mulmeong/utility/common/response/BaseResponseStatus.java

Cualone committed:
:sparkles: feat: 특정 유저의 팔로워 목록 조회

Cualone committed:
:sparkles: feat: 특정 유저의 팔로잉 목록 조회

SeongGwangJu commented:
현재 #10 #15 #13 #14 순으로 PR봐달라고 하셨는데,
리뷰하려고 보면 모든 PR이 이전 PR들의 커밋을 포함하는 형식으로 되어있어 매 리뷰마다 반복적으로 동일한 내용을 봐야하는 문제가 생깁니다.

다음부터는 1)가급적 PR은 독립적으로 분기되도록하고, 
PR 리뷰가 늦어질 경우 2)작성해놓은 PR에 추가커밋/ PR내용 수정을하거나, 또는 3)순차적으로 PR을 게시하는게 좋을 것 같습니다!

SeongGwangJu reviewed (commented):
현재 #10 #15 #13 #14 순으로 PR봐달라고 하셨는데,
리뷰하려고 보면 모든 PR이 이전 PR들의 커밋을 포함하는 형식으로 되어있어 매 리뷰마다 반복적으로 동일한 내용을 봐야하는 문제가 생깁니다.

다음부터는 1)가급적 PR은 독립적으로 분기되도록하고,
PR 리뷰가 늦어질 경우 2)작성해놓은 PR에 추가커밋/ PR내용 수정을하거나, 또는 3)순차적으로 PR을 게시하는게 좋을 것 같습니다!

SeongGwangJu commented:
또한 커밋 로그에서 Merge 커밋이 많이 보이는데, 번거롭게 커밋도 남겨야하고(심지어 깃모지까지 더해서), 히스토리가 남아 추적이 어려운 문제가 발생합니다.  Pull로 가져올때  git pull --rebase로 가져오는 것을 권장드립니다..

gyudol commented on src/main/java/com/mulmeong/utility/adapter/out/infrastructure/mongo/repository/FollowRepository.java:
pageNo가 0 고정인 것으로 보이는데, 이 경우에 데이터가 많을 때 값이 적절하게 잘 나오는지 궁금합니다

Cualone replied:
수정 완료

Cualone committed:
:twisted_rightwards_arrows: merge: Merge branch 'develop' of https://github.com/mul-meong/utility into feature/follow/#11

# Conflicts:
#	src/main/java/com/mulmeong/utility/UtilityApplication.java
#	src/main/java/com/mulmeong/utility/adapter/in/web/controller/DislikeController.java
#	src/main/java/com/mulmeong/utility/adapter/in/web/controller/LikesController.java

Cualone committed:
:bug: fix: 좋아요 목록 페이지네이션 pageNo 추가

Cualone committed:
:bug: fix: 팔로워, 팔로잉 페이지네이션 pageNo 추가

Cualone committed:
:bug: fix: response pageNo 추가

gyudol reviewed (approved):
고생하셨습니다 📦️
